k8s/profiles.go

```diff
@@ -0,0 +1,249 @@
+package k8s
+
+import (
+	"context"
+	"reflect"
+	"strings"
+
+	v1 "github.com/openfaas/faas-netes/pkg/apis/openfaas/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	typedCorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const ProfileAnnotationKey = "com.openfaas.profile"
+
+// ProfileClient defines the interface for CRUD operations on profiles
+// and applying faas-netes profiles to function Deployments.
+type ProfileClient interface {
+	Get(ctx context.Context, namespace string, names ...string) ([]Profile, error)
+}
+
+// Profile is and openfaas api extensions that can be predefined and applied
+// to functions by annotating them with `com.openfaas.profile: name1,name2`
+type Profile v1.ProfileSpec
+
+type profileCMClient struct {
```

Comment:
`profileCMClient` -> `profileConfigMapClient` would be less cryptic and help with maintenance / setting up to speed with the code as a new-comer